spec/support/lib/module_validation.rb

```diff
@@ -145,6 +159,19 @@ def validate_target_platforms
       end
     end
 
+    def validate_attack_reference_format
```

Comment:
Tested and works as expected:
```
  1) modules it should behave like all modules with module type can be instantiated exploit windows/smb/psexec behaves like a module with valid metadata verifies modules metadata
     Failure/Error: expect(validator.errors.full_messages).to be_empty
       expected `["References ATT&CK reference 'Z1059' is invalid. Must start with one of [\"TA\", \"DS\", \"S\", \"M\", \"A\", \"G\", \"C\", \"T\"] and be followed by digits/periods, no whitespace."].empty?` to be truthy, got false
       ```
       
Job: https://github.com/rapid7/metasploit-framework/actions/runs/15586996464/job/43897873928?pr=20289

Reply:
Added in https://github.com/rapid7/metasploit-framework/pull/20289/commits/b5fcaca940449e0133b44ab4046bbeea1e4c2e15.

Reply:
It'd be great to add a unit test example for this to lock the behavior in :+1: